### PR TITLE
backport: bitcoin#18875,11413,16946,17824,18601,18585,20271,19871,20039

### PR DESCRIPTION
**Pull request by vijaydasmp**

### CODE REVIEW

**Comment by vijaydasmp** [comment]
evo/cbtx.cpp:401:25: error: comparison of **integer expressions of different signedness: ‘int’ and ‘unsigned int’** [-Werror=sign-compare]
  401 |         if (curCLHeight < prevCLHeight) {



wallet/wallet.cpp:3582:32: error: **comparison of integer expressions of different signedness: ‘int’ and ‘const unsigned int**’ [-Werror=sign-compare]
 3582 |                     if (nBytes > MAX_STANDARD_TX_SIZE) {

**Comment by vijaydasmp** [comment]
Hello @UdjinM6  @knst  please review

File: doc/release-notes-11413.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+----------------------
+
+The `bumpfee`, `fundrawtransaction`, `sendmany`, `sendtoaddress`, and `walletcreatefundedpsbt`
+RPC commands have been updated to include two new fee estimation methods "BTC/kB" and "sat/B".
+The target is the fee expressed explicitly in the given form. Note that use of this feature
+will trigger BIP 125 (replace-by-fee) opt-in.
+
+In addition, the `estimate_mode` parameter is now case insensitive for all of the above RPC commands.
+
+The `bumpfee` command now uses `conf_target` rather than `confTarget` in the options.
```
Comment (UdjinM6) [inline doc/release-notes-11413.md:11]: 11413: we don't have bip125/rbf/`bumpfee`
```suggestion
The `fundrawtransaction`, `sendmany`, `sendtoaddress`, and `walletcreatefundedpsbt`
RPC commands have been updated to include two new fee estimation methods "DASH/kB" and "duff/B".
The target is the fee expressed explicitly in the given form.

In addition, the `estimate_mode` parameter is now case insensitive for all of the above RPC commands.
```

File: src/policy/feerate.h [hunk 2]
```diff
@@ -11,7 +11,17 @@
 
 #include <string>
 
-extern const std::string CURRENCY_UNIT;
+const std::string CURRENCY_UNIT = "DASH"; // One formatted unit
+const std::string CURRENCY_ATOM = "duffs"; // One indivisible minimum value unit
```
Comment (UdjinM6) [inline src/policy/feerate.h:15]: 11413:
```suggestion
const std::string CURRENCY_ATOM = "duff"; // One indivisible minimum value unit
```

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -51,6 +51,8 @@ using interfaces::FoundBlock;
 
 static const std::string WALLET_ENDPOINT_BASE = "/wallet/";
 
+static const uint32_t WALLET_DASH_KB_TO_DUFFS_B = COIN / 1000;
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp:54]: 11413:
```suggestion
static const uint32_t WALLET_DASH_KB_TO_DUFFS_B = COIN / 1000; // 1 duff / B = 0.00001 DASH / kB
```

File: src/policy/feerate.h [hunk 4]
```diff
@@ -11,7 +11,17 @@
...
+
+/* Used to determine type of fee estimation requested */
+enum class FeeEstimateMode {
+    UNSET,        //!< Use default settings based on other criteria
+    ECONOMICAL,   //!< Force estimateSmartFee to use non-conservative estimates
+    CONSERVATIVE, //!< Force estimateSmartFee to use conservative estimates
+    DASH_KB,       //!< Use explicit DASH/kB fee given in coin control
+    DUFFS_B,        //!< Use explicit sat/B fee given in coin control
```
Comment (UdjinM6) [inline src/policy/feerate.h:23]: 11413:
```suggestion
    DASH_KB,      //!< Use explicit DASH/kB fee given in coin control
    DUFFS_B,      //!< Use explicit duff/B fee given in coin control
```

File: src/wallet/rpcwallet.cpp [hunk 5]
```diff
@@ -3254,6 +3274,12 @@ void FundTransaction(CWallet* const pwallet, CMutableTransaction& tx, CAmount& f
 
         if (options.exists("feeRate"))
         {
+            if (options.exists("conf_target")) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Cannot specify both conf_target and feeRate");
+            }
+            if (options.exists("estimate_mode")) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Cannot specify both estimate_mode and feeRate");
+            }
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp:3282]: 11413:
missing https://github.com/bitcoin/bitcoin/pull/11413/files#diff-63c7710d72381cffe692175dd70e47f1891bf75c3e09249ed826f9df3917d02dR3021 below

File: test/functional/wallet_basic.py [hunk 6]
```diff
@@ -223,7 +223,60 @@ def run_test(self):
...
+            conf_target=fee_duffs_per_b,
+            estimate_mode='duffs/b',
+        )
+        self.nodes[2].generate(1)
+        self.sync_all(self.nodes[0:3])
+        balance = self.nodes[2].getbalance()
+        node_2_bal = self.check_fee_amount(balance, node_2_bal - Decimal('10'), explicit_fee_per_byte, count_bytes(self.nodes[2].gettransaction(txid)['hex']))
+        assert_equal(balance, node_2_bal)
+        node_0_bal += Decimal('10')
+        assert_equal(self.nodes[0].getbalance(), node_0_bal)
```
Comment (UdjinM6) [inline test/functional/wallet_basic.py:278]: 11413: missing second part of tests https://github.com/bitcoin/bitcoin/pull/11413/files#diff-047f185450ff2093df390b522db9b6aa1c87353abf7e7a99c00390db8c707f6eR406

File: depends/README.md [hunk 7]
```diff
@@ -112,6 +112,10 @@ The following can be set when running make: `make FOO=bar`
 <dd>Don't download/build/cache packages needed for enabling upnp</dd>
 <dt>NO_NATPMP</dt>
 <dd>Don't download/build/cache packages needed for enabling NAT-PMP</dd>
+<dt>ALLOW_HOST_PACKAGES</dt>
+<dd>Packages that are missed in dependencies (due to `NO_*` option or
+build script logic) are searched for among the host system packages using
+`pkg-config`. It allows building with packages of other (newer) versions</dd>
```
Comment (UdjinM6) [inline depends/README.md:118]: 19124: we don't have `ALLOW_HOST_PACKAGES` option

File: test/functional/mempool_accept.py [hunk 8]
```diff
@@ -4,6 +4,8 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 """Test mempool acceptance of raw transactions."""
 
+
+from decimal import Decimal
```
Comment (UdjinM6) [inline test/functional/mempool_accept.py:8]: 19940: this is like super-partial :D  the fix should be simply a part of 07c6897cae1875c7ad88bc3127bf26541b74c8ff (20039 backport)

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/functional/mempool_accept.py (hunk 8 above)
Comment (vijaydasmp) [reply test/functional/mempool_accept.py:8]: :) Sure will make it a part of 20039 ,

**Comment by UdjinM6** [comment]
pls see 662d1df582d3788d2d7d5a93e7d30dba398a2ad0

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by knst — APPROVED** [review]
utACK

Although, need to remove 18216 and 19124 from title of PR

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**